### ISSUE TITLE
Feature request: Add confidence scores to library output(s)

### ISSUE THREAD

**Issue by Allypost** [robertknight/ocrs#101]
It would be really nice to have some sort of (maybe even optional) confidence scores for text recognition output.

It seems that information is already present in the lower-level libraries so it would be just a matter of exposing and/or normalizing it for the library user.

**Comment by robertknight** [comment]
For the default [`DecodeMethod::Greedy`](https://docs.rs/ocrs/latest/ocrs/enum.DecodeMethod.html) decoding of the neural network outputs there is a score per character that could be exposed. If the beam search decoding method is used, it gets a bit more complex because there is a score per line which is a sum over multiple "paths" through the outputs. The raw scores the network computes are log probabilities (output of a [`LogSoftmax`](https://pytorch.org/docs/stable/generated/torch.nn.LogSoftmax.html) operation). It might be more user friendly to convert them back to probabilities first.
